llvm/test/Transforms/Attributor/value-simplify.ll

```diff
@@ -1422,7 +1417,7 @@ define i1 @user_of_not_called() {
 ; CHECK-LABEL: define {{[^@]+}}@user_of_not_called() {
 ; CHECK-NEXT:    call void @useFnDecl(ptr addrspace(42) noundef nonnull addrspacecast (ptr @not_called1 to ptr addrspace(42)))
 ; CHECK-NEXT:    call void @useFnDef(ptr addrspace(42) noundef nonnull addrspacecast (ptr @not_called2 to ptr addrspace(42)))
-; CHECK-NEXT:    ret i1 icmp eq (ptr addrspace(42) addrspacecast (ptr @not_called3 to ptr addrspace(42)), ptr addrspace(42) null)
+; CHECK-NEXT:    ret i1 false
```

Comment:
This change looked suspicious to me -- I was afraid that we started to simplify a null comparison through an addrspace cast, which wouldn't generally be valid. But this doesn't seem to be the case as far as ConstantFolding/InstSimplify are concerned. This is some Attributor-specific behavior and it has been flipping back and forth in this test in multiple previous changes. Maybe @jdoerfert knows what is happening here.